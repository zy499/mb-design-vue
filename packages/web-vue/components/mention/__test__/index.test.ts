import { mount } from '@vue/test-utils';
import Mention from '../index';

describe('Mention', () => {
  test('mention correctly', async () => {
    const wrapper = mount(Mention, {
      props: {
        data: ['Bytedance', 'Bytedesign', 'Bytenumner'],
      },
    });
    const input = wrapper.find('input');
    await input.trigger('focusin');
    await input.setValue('@');

    expect(
      wrapper.findComponent({ name: 'SelectDropdown' }).html()
    ).toMatchSnapshot();
  });

  test('should select value', async () => {
    const wrapper = mount(Mention, {
      props: {
        data: ['Bytedance', 'Bytedesign', 'Bytenumner'],
      },
    });
    const input = wrapper.find('input');
    await input.trigger('focusin');
    await input.setValue('@');
    const dropdown = wrapper.findComponent({ name: 'SelectDropdown' });
    await input.trigger('keydown', { key: 'ArrowDown' });
    expect(dropdown.find('.mb-select-option-active').text()).toBe('Bytedesign');
    await input.trigger('keydown', { key: 'ArrowUp' });
    expect(dropdown.find('.mb-select-option-active').text()).toBe('Bytedance');
    await input.trigger('keydown', { key: 'Enter' });
    expect(wrapper.emitted('change')?.[1]).toEqual(['@Bytedance']);
  });
});
