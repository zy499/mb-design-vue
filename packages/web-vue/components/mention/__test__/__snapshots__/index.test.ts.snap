// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mention mention correctly 1`] = `
"<div class=\\"mb-select-dropdown\\">
  <!--v-if-->
  <!--v-if-->
  <div class=\\"mb-select-dropdown-list-wrapper\\">
    <ul class=\\"mb-select-dropdown-list\\">
      <li class=\\"mb-select-option mb-select-option-active\\">
        <!--v-if--><span class=\\"mb-select-option-content\\">MaBang</span>
        <!--v-if-->
      </li>
      <li class=\\"mb-select-option\\">
        <!--v-if--><span class=\\"mb-select-option-content\\">Bytedesign</span>
        <!--v-if-->
      </li>
      <li class=\\"mb-select-option\\">
        <!--v-if--><span class=\\"mb-select-option-content\\">Bytenumner</span>
        <!--v-if-->
      </li>
    </ul>
  </div>
  <!--v-if-->
</div>"
`;
